### PR TITLE
Let each crier controller have its own queue

### LINKED ISSUE #10602: Crier sometimes didn't report?
/assign
/kind bug

@rancerbeta found some instances, seems the informer added the prowjobs but it didn't want to report.
need to add more loggings to investigate

### PR DESCRIPTION
**Pull request by krzyzacy**
orz... it always works when I debugging it with one controller... but screwey when I have multiple ones (gerrit + pubsub...) because all the keys are added twice (deduped by workqueue) and forgot once and :bomb: 

/pony dumb me

fixes #10602
/assign @fejta @stevekuznetsov @munnerz 
cc @rancerbeta

### CODE REVIEW

**Comment by krzyzacy** [comment]
:-\ I'll write more tests I promise

**Comment by munnerz** [comment]
😄

/lgtm
/hold

**Comment by munnerz** [comment]
(in case anyone else wants to say anything - feel free to unhold!)

File: prow/cmd/crier/main.go [hunk 1]
```diff
@@ -159,7 +157,7 @@ func main() {
 			controllers,
 			crier.NewController(
 				prowjobClient,
-				queue,
+				kube.RateLimiter(controllerName),
```
Comment (munnerz) [inline prow/cmd/crier/main.go]: Hm - the controllerName should be unique between all the queues. Assuming this gets plumbed down as the name of the workqueue...

Comment (krzyzacy) [reply prow/cmd/crier/main.go]: 👍 good catch

Comment (krzyzacy) [reply prow/cmd/crier/main.go]: done

**Pull request merged**